test/rules/dangling_library_doc_comments_test.dart

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:test_reflective_loader/test_reflective_loader.dart';
+
+import '../rule_test_support.dart';
+
+main() {
+  defineReflectiveSuite(() {
+    defineReflectiveTests(DanglingLibraryDocCommentsTest);
+  });
+}
+
+@reflectiveTest
+class DanglingLibraryDocCommentsTest extends LintRuleTest {
```

Comment:
I'll suggest one additional test (besides the ones that might be implied by other comments):
```
/// Library comment

/// Class comment
class C {}
```

I think there should be a lint here but suspect that there won't be because I don't think we look above the class' documentation comment.

Reply:
Ah ha good catch. Done.